test/stdlib/TestXPCEncoder.swift

```diff
@@ -0,0 +1,552 @@
+}
+
+/// A derived class
+fileprivate class Programmer : Person {
+    let favoriteIDE: String
+
+    init(name: String, email: String, website: URL? = nil, favoriteIDE: String) {
+        self.favoriteIDE = favoriteIDE
+        super.init(name: name, email: email, website: website)
+    }
+
+    required init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        favoriteIDE = try container.decode(String.self, forKey: .favoriteIDE)
+        try super.init(from: container.superDecoder())
+    }
+
+    override func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: CodingKeys.self)
+        try container.encode(favoriteIDE, forKey: .favoriteIDE)
+        try super.encode(to: container.superEncoder())
+    }
+
+    enum CodingKeys : String, CodingKey {
+        case favoriteIDE
+    }
+
+    override func isEqual(_ other: Person) -> Bool {
+        if let programmer = other as? Programmer {
+            guard self.favoriteIDE == programmer.favoriteIDE else { return false }
+        }
+
+        return super.isEqual(other)
+    }
+
+    override class var testValue: Programmer {
+        return Programmer(name: "Johnny Appleseed", email: "appleseed@apple.com", favoriteIDE: "XCode")
+    }
+}
+
+/// A class which shares its encoder and decoder with its superclass.
+fileprivate class Employee : Person {
+  let id: Int
+
+  init(name: String, email: String, website: URL? = nil, id: Int) {
+    self.id = id
+    super.init(name: name, email: email, website: website)
+  }
+
+  enum CodingKeys : String, CodingKey {
+    case id
+  }
+
+  required init(from decoder: Decoder) throws {
+    let container = try decoder.container(keyedBy: CodingKeys.self)
+    id = try container.decode(Int.self, forKey: .id)
+    try super.init(from: decoder)
+  }
+
+  override func encode(to encoder: Encoder) throws {
+    var container = encoder.container(keyedBy: CodingKeys.self)
+    try container.encode(id, forKey: .id)
+    try super.encode(to: encoder)
+  }
+
+  override func isEqual(_ other: Person) -> Bool {
+    if let employee = other as? Employee {
+      guard self.id == employee.id else { return false }
+    }
+
+    return super.isEqual(other)
+  }
+
+  override class var testValue: Employee {
+    return Employee(name: "Johnny Appleseed", email: "appleseed@apple.com", id: 42)
+  }
+}
+
+/// A simple company struct which encodes as a dictionary of nested values.
+fileprivate struct Company : Codable, Equatable {
+    let address: Address
+    var employees: [Employee]
+
+    init(address: Address, employees: [Employee]) {
+        self.address = address
+        self.employees = employees
+    }
+
+    static func ==(_ lhs: Company, _ rhs: Company) -> Bool {
+        return lhs.address == rhs.address && lhs.employees == rhs.employees
+    }
+
+    static var testValue: Company {
+        return Company(address: Address.testValue, employees: [Employee.testValue])
```

Comment:
I would sleep better if it had more than one value in the `employees` array. Just in case...